core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java

```diff
@@ -1070,51 +1085,89 @@ public void addListener(Object listener, KeyFilter<? super K> filter, ClassLoade
       }
 
       // If we have a segment listener handler, it means we have to do initial state
-      QueueingSegmentListener handler = segmentHandler.remove(generatedId);
+      QueueingSegmentListener<K, V, ? extends Event<K, V>> handler = segmentHandler.remove(generatedId);
       if (handler != null) {
          if (trace) {
             log.tracef("Listener %s requests initial state for cache", generatedId);
          }
-         CacheStream entryStream = cache.wired().withEncoding(keyConversion.getEncoderClass(), valueConversion.getEncoderClass()).cacheEntrySet().stream();
-         Stream<CacheEntry<K, V>> usedStream = entryStream.segmentCompletionListener(handler);
+         Queue<IntermediateOperation> intermediateOperations = new ArrayDeque<>();
```

Comment:
`Queue` suggests that items are being removed from the front, but AFAICT it's only iterated, so I'd replace it with a `List`.

Reply:
tbh just Collection would work. I believe this originates from old DistributedStream code where I used a Queue and removed first elements at some point. This is no longer true though.